components/gitpod-cli/cmd/tasks-attach.go

```diff
@@ -97,25 +94,29 @@ var attachTaskCmd = &cobra.Command{
 				default:
 					fmt.Println(e.Code(), e.Message())
 				}
-				return
+				return nil
 			} else {
-				panic(err)
+				return err
 			}
 		}
 		ppid := int64(os.Getppid())
 
 		if ppid == terminal.Pid {
 			fmt.Println("You are already in terminal:", terminalAlias)
-			return
+			return GpError{OutCome: utils.UserErrorCode, ErrorCode: utils.UserErrorCode_AlreadyAttached}
 		}
 
 		interactive, _ := cmd.Flags().GetBool("interactive")
 		forceResize, _ := cmd.Flags().GetBool("force-resize")
 
-		client.AttachToTerminal(context.Background(), terminalAlias, supervisor.AttachToTerminalOpts{
+		exitCode, err := client.AttachToTerminal(cmd.Context(), terminalAlias, supervisor.AttachToTerminalOpts{
 			ForceResize: forceResize,
 			Interactive: interactive,
 		})
+		if err != nil {
```

Comment:
@iQQBot I got system erro by typing `exit` in attached terminal. It should be success in this case?

i.e. create VS Code terminal do `gp tasks attach` and then run `exit`